thorlcr/master/thmastermain.cpp

```diff
@@ -742,11 +756,13 @@ int main( int argc, const char *argv[]  )
         unsigned gmemSize = globals->getPropInt("@globalMemorySize"); // in MB
         if (0 == gmemSize)
         {
+            // NB: This could be in a isContainerized(), but the 'workerResources' section only applies to containerized setups
             const char *workerResourcedMemory = globals->queryProp("workerResources/@memory");
             if (!isEmptyString(workerResourcedMemory))
             {
                 offset_t sizeBytes = friendlyStringToSize(workerResourcedMemory);
                 gmemSize = (unsigned)(sizeBytes / 0x100000);
+                gmemSize = gmemSize * containerRoxieMemPC / 100;
```

Comment:
Feels like there is a danger of overflow here?

Reply:
I think it's okay, gmemSize is MB at this stage so even with containerRoxieMemPC at 100, gmemSize can be up to 42949672 MB